lfric_macros/apply_macros.py

```diff
@@ -207,6 +207,48 @@ def banner_print(message):
     print(f"\n{(len(message)+4)*'*'}\n* {message} *\n{(len(message)+4)*'*'}\n")
 
 
+def read_example_meta_source(path):
+        """
+        Read the metadata source from an 'example_meta_source.txt' file
+        Return:
+            - str, the name of the metadata source
+        """
+
+        with open(os.path.join(path, "example_meta_source.txt")) as f:
+            return f.readlines.strip()
+
+
+def create_temporary_config_file(example_path, meta_name, last_tag):
+    """
+    Create a temporary rose-app.conf file from an example configuration.nml
+    Use the 'rose namelist-dump' command
+    Add a file:configuration.nml section
+    Add a meta= import to the top
+    Returns:
+        - path to this file in a temporary directory
+    """
+    tempdir = tempfile.mkdtemp()
+    tempconf = os.path.join(tempdir, "rose-app.conf")
+    command = f"rose namelist-dump {example_path} -o {tempconf}"
+    result = run_command(command)
+    if result.returncode:
+        raise Exception(
+            f"Failed to run 'rose namelist-dump' on nml file in {example_path}."
+            f"\nThe error message produced was:\n\n{result.stder}"
+        )
+
+    with open(tempconf) as f:
+        lines = f.readlines()
+
+    # Add the metadata import statement
+    # lines = [f"meta={meta_name}/{last_tag}\n\n"] + lines
+
+    with open(tempconf, "w") as f:
+        f.write("".join(x for x in lines))
```

Comment:
I think the below should let you read and write without having to open and close the file twice.

```suggestion
    # Add the metadata import statement
    # lines = [f"meta={meta_name}/{last_tag}\n\n"] + lines
    with open(tempconf, "r+") as f:
        lines = f.readlines()
        f.seek(0)
        f.write("".join(x for x in lines))        
```